99.md

```diff
@@ -0,0 +1,170 @@
+
+For a general overview on what prediction markets are, see Appendix A.
+
+### Creation of predictable events
+
+To create a predictable event, `kind 1099` events are used as defined by NIP-01 like this:
+
+```json
+{
+  "content": <description about what should be predicted>,
+  "tags": [
+    ["outcome": <description of possible outcome>],
+    ["outcome": <description of another possible outcome>],
+    ...
+  ],
+  "kind": 1099,
+  "id": <event_id>
+  "pubkey": <pubkey of creator>,
+  "created_at": <timestamp>,
+  "sig": <signature>
+}
+```
+
+Example for a binary option event:
+
+```json
+{
+  "content": "Will Twitter go bankrupt in 2023?",
+  "tags": [
+    ["outcome": "Yes"],
+    ["outcome": "No"]
+  ],
+  "kind": 1099,
+  ...
+}
+```
+
+Exchanges run clients which have subscribed to these events.
+Per received event, they create parametrized replaceable events as defined by NIP-33 per possible outcome:
+
+<!--
+Working on a better explanation. Exchanges don't have to provide services for every event.
+
+Also, I want to provide an example flow from start to finish.
+This example flow should especially answer the following questions regarding "market bootstrapping":
+  Q: How are initial prices set?
+  A: Without any participants, there is no information regarding odds so every outcome is equally probable. So every share should cost the same.
+
+  Q: What happens if there will only be one participant? Do they not always lose money because of potential commission fees?
+  A: ???
+
+  Q: Why would the first participant buy more than a single share? If they buy multiple shares, the next participant has an advantage because the next participant can just buy a single share to profit the most with the least risk (assuming no other participant will follow).
+  Basically this: Why would you bet 100 sats to win 100 sats if you can just bet 1 sat to win 100 sats?
+  A: ??? Maybe I am just confused with shares vs. odds. I need to learn more about prediction markets first.
+-->
+
+```json
+{
+  "content": <outcome description>,
+  "tags": [
+    ["e", <id of predictable event>],
+    ["zap", <id of predictable event>+<outcome index>@SERVICE_URL, "lud16"]
+    ["d", <id of predictable event>+<outcome index>]
+    ["odds", <odds>, <odds format>]
+  ],
+  "kind": 301099,
+  "id": <outcome event id>,
+  "pubkey": <pubkey of bookmaker>,
+  "created_at": <timestamp>
+  ...
+}
+```
+
+These events include `zap` tags to link zapping the event with buying a share of that specific outcome with the current given odds.
+
+<!-- how to sell shares using NIP57 ??? -->
+
+Outcomes are specified by using plus addresses and zero-based indexing the `outcome` tags in the referred predictable event. Participants must validate that the plus address local part references an outcome with the same description as in `content` to prevent ambiguity.
+
+<!-- plus addresses are not supported by LUD16 ... yet? -->
```

Comment:
> The idea here is that a SERVICE can offer human-readable addresses for users or specific internal endpoints that use the format <username>@<domainname>, e.g. [satoshi@bitcoin.org](mailto:satoshi@bitcoin.org). A user can then type these on a WALLET. The <username> is limited to a-z0-9-_.. Please note that this is way more strict than common email addresses as it allows fewer symbols and only lowercase characters.

-- https://github.com/lnurl/luds/blob/luds/16.md

Can a `+` be added as a valid character to allow plus addressing? This would be backwards compatible.

If not, I need to find another way to refer to specific outcomes in an event within a LN address. I could just use `_` instead of `+`, I guess. But I liked how plus addressing nicely fits the use case here.